packages/sdk/src/util/webHost.ts

```diff
@@ -67,11 +75,8 @@ export class WebHost {
 			// host static binaries
 			(req, res, next) => this.serveStaticBuffers(req, res, next),
 			// host static files
-			Restify.plugins.serveStatic({
-				directory: this._baseDir,
-				default: 'index.html'
-			}
-			));
+			Restify.plugins.serveStaticFiles(this._baseDir, { index: 'index.html', etag: true }),
+			Restify.plugins.conditionalRequest());
```

Comment:
It isn't clear to me how Restify would be able to check the static buffer's etag, and return a 304/Not Modified if the resource hasn't changed. How does that work?

Reply:
I compute the etag when the buffer is registered, then serve it with requests. That `conditionalRequest` plugin doesn't compute etags, it just looks at the header already set in previous steps.

Reply:
If the app re-registers an updated buffer, it will get a new etag. Does this invalidate the cached headers in the plugin?

Reply:
Nothing is cached anywhere inside Restify. The only caching is what you see. That said, this implementation was totally broken, and I've rewritten it. TL;DR: Nothing that happens after `res.send` counts, and the conditional request middleware was after the main handler. I split it up.

Reply:
Something I'm still not clear on: When a GET request comes in for a static buffer, and the request includes an etag (via If-None-Match header), how exactly is the incoming etag compared with the static buffer's etag in order to know whether to send a 304/Not Modified, or send a 200/OK + the asset?

Reply:
So there are two routes here, static files and static buffers. Static files have always supported these headers it seems. For static buffers, I had to modify it so the new request flow goes like this:

1. Request for a buffer comes in and hits the `/buffers/:name` route.
2. The first handler in the chain, `checkStaticBuffers`, looks up whether the given buffer exists, and if so, adds the current ETag value to the response, but does not yet *send* the response. This ETag value is computed from the contents of the buffer when it's registered.
3. The second handler, `Restify.plugins.conditionalRequest`, checks the response ETag header against any `If-Match` request headers, and as required sends the `304` response and terminates the handler chain.
4. The third and final handler, `serveStaticBuffers`, is only reached if a full response is required. It sends the remaining response headers and the response body, and closes out the request.

Reply:
Thanks for the explanation. This sounds like the right flow.